src/OpenTracing/NoopTracer.php

```diff
@@ -12,24 +13,21 @@ public static function create()
         return new self();
     }
 
-    /** @return Span */
     public function startSpan(
         $operationName,
         SpanReference $parentReference = null,
         $startTimestamp = null,
         Tag ...$tags
     ) {
-        // TODO: Implement startSpan() method.
+        return NoopSpan::create();
```

Comment:
The reason why I usually use static `create` methods is because I believe that an object can have different ways to create (for instance, in a real implementation of `Span` you can have `createAsChildOf` and `createAsFollowsFrom`. To me, `__construct` methods are more an straightforward way to create objects and they should not include business rules on creation whereas a `create` method can include validation rules. Regarding the Liskov's substitution principle I did not add the `create` method on the interface for this reason and honestly I prefer to use `final` clases for this VOs because I believe there's no reason to extend this class since you have the interface (O from `SOLID`) and if one really want to extend the functionality I'd use decoration (like on https://github.com/jcchavezs/dd-opentracing-php/blob/master/src/DdOpenTracing/Span.php).